Eventbrite Block: Update saved block markup

Copons opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #14506

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* When the Eventbrite uses the Button & Modal embed type, add `.wp-block-button` to the block wrapper in order to appropriately style the `.wp-block-button__link` inside.
* Change the markup to be `<a role="button">` rather than `<button>` to be compatible with WordPress.com `wp_kses` configuration.

| Before | After |
| - | - |
| <img width="629" alt="Screen Shot 2020-01-28 at 19 54 56" src="https://user-images.githubusercontent.com/1699996/73321453-194efd00-4208-11ea-847f-ef3cd4367a0e.png"> | <img width="555" alt="Screenshot 2020-01-29 at 13 50 01" src="https://user-images.githubusercontent.com/2070010/73362588-01768800-429f-11ea-8cc2-61b13d003cb8.png"> |

⚠️ Note: this is change breaks existing blocks. Since the block is not released yet, I haven't created a deprecation/migration path. Just keep in mind that you might need to let Gutenberg try to automatically fix any existing blocks, or just remove them and add new ones.

**Additional Context**

In Core Guten, `wp-block-button__link` is the inner element inside a `wp-block-button`.
So, it's to be expected that it malfunctions when used "outside" `wp-block-button`.

E.g. this is how a normal Button block is rendered in the editor:
<img width="921" alt="Screenshot 2020-01-29 at 13 31 50" src="https://user-images.githubusercontent.com/2070010/73360952-e0606800-429b-11ea-83dd-ee617fde9fd5.png">

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Bug fix

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Open the Editor, insert an Eventbrite block, and provide an Eventbrite event URL.
* From the block sidebar, change the Embed Type to "Button & Modal".
* Make sure the previewed button style is consistent to the current theme, instead of being dark grey with black text color.
* Save and view the post.
* Make sure the rendered button is identical to how it looks in the editor.
* Modify the button style from the block sidebar (make it outline, change the colors and the border radius).
* Make sure the button is updated accordingly in both the editor and the front end.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* N/A

Copons committed:
Make sure wp-block-button__link is used inside wp-block-button

Copons committed:
Try using a simple link instead of a button

Copons commented:
I've added a new commit that fixes an issue caught by @creativecoder.
On WPCOM, Eventbrite blocks in button mode weren't displayed on the front end.

As it turns out, they were, but stripped of the `<button>` tag, which not only doesn't work, but also sometimes turns the text color the same as the background color, making it invisible.

My best guess is that WPCOM's `kses` strips the `button` tag while Core doesn't, reason why the block worked fine in our local environments.

In 3ecf911 I've simply changed the `RichText` output from `button` to `a`.
It works for me on both Jetpack and Simple, but please double check for possible regressions.

Copons committed:
Add a role="button"

Copons commented:
For more context:

The original approach of outputting a `button` element when the Eventbrite block is of type "Button & Modal" doesn't work on WPCOM because `button` is not whitelisted in the KSES.
At the same time, an `a` element that pretends to be a `button` is [not very accessible](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Accessibility).

As a middle ground we are adding a `role="button"`, which should be OK, if not for another issue.
[Button roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) are supposed to be focusable (and anchors are), have an `onClick` handler (and Eventbrite adds it), and also an `onKeyDown` for <kbd>Enter</kbd> and <kbd>Space</kbd> (and Eventbrite **doesn't** add them).

I think it's a minor issue, and not one we are able to solve cleanly.

This said, I'm not an a11y expert by any means, so let's discuss this further!

creativecoder committed:
Adds native event handlers for button behavior on <a> modal trigger

creativecoder commented:
I verified that VoiceOver sees the `<a role=button>` element as a button, so I think this has what it needs for screen reader accessibility.

For wai-aria recommended button behavior, I pushed a commit with some inline javascript to add event handlers. It feels old school, but simple enough that it should be okay--curious what you think about this approach @Copons

Copons commented on extensions/blocks/eventbrite/save.js:
```diff
@@ -57,15 +59,18 @@ function saveButton( eventId, attributes ) {
+				role="button"
 				style={ buttonStyle }
+				tagName="a"
+				href={ url }
```
Nice one @creativecoder, love that you got to remove the `noscript` with such a simple change! ✨

Copons committed:
Add target blank to the button

Copons committed:
Adjust inline scripts spacing

Copons commented:
@creativecoder I think the workaround is as good as we could pull off, and tests very well for me with and without JS.

I've added a `target="_blank"` to the button, like it was in the `noscript` version, and cleaned up the spacing of the inline scripts.

Copons committed:
Add also missing rel attribute

creativecoder committed:
Check for event.target before clicking it

Prevents some console errors in the editor

creativecoder committed:
Merge branch 'master' into fix/14506-eventbrite-button-color

creativecoder committed:
Remove redundant link from Modal & Button saved markup

creativecoder reviewed (approved):
@Copons I merged in master so I could test the latest version of the block and made a couple of small tweaks.

This looks good, working for me expected from my local Jetpack site and from wpcom, I think it's ready to go.

jeherve reviewed (approved):
This looks good to me, it tests well! 🚢

creativecoder commented:
r202396-wpcom

jeherve commented:
Cherry-picked to `branch-8.2` in 1fa4828d371dc5d525b578313b24f805bfb59f54
